minimesos/src/main/java/com/containersol/minimesos/util/EnvironmentBuilder.java

```diff
@@ -0,0 +1,31 @@
+package com.containersol.minimesos.util;
+
+import java.util.Map;
+import java.util.TreeMap;
+
+/**
+ * Provides convenient API for building a map of environment variables.
+ * Produces the String[] format needed by CreateContainerCmd.setEnv()
+ */
+public class EnvironmentBuilder {
```

Comment:
Added unit test for single complex merge scenario, let me know if this is good enough.